openvdb/openvdb/tree/LeafManager.h

```diff
@@ -635,34 +615,99 @@ class LeafManager
 
 private:
 
-    // This a simple wrapper for a c-style array so it mimics the api
-    // of a std container, e.g. std::vector or std::deque, and can be
-    // passed to Tree::getNodes().
-    struct MyArray {
-        using value_type = LeafType*;//required by Tree::getNodes
-        value_type* ptr;
-        MyArray(value_type* array) : ptr(array) {}
-        void push_back(value_type leaf) { *ptr++ = leaf; }//required by Tree::getNodes
-    };
-
-    void initLeafArray()
+    void initLeafArray(bool serial = false)
     {
-        const size_t leafCount = mTree->leafCount();
+        // Build an array of all nodes that have leaf nodes as their immediate children
+
+        using NodeChainT = typename NodeChain<RootNodeType, RootNodeType::LEVEL>::Type;
+        using NonConstLeafParentT = typename NodeChainT::template Get</*Level=*/1>;
```

Comment:
Guess this isn't guaranteed to be the leaf node parent! Should probably be:
`
using NonConstLeafParentT = typename NodeChainT::PopBack::Back;
`